File: code/community/1306402371709964289/deepgram-diarization-pricing.py
import os
import requests

# Load environment variables
api_key = os.getenv('DEEPGRAM_API_KEY')
audio_url = os.getenv('AUDIO_URL')

# API endpoint
url = "https://api.deepgram.com/v1/listen"

# Headers
headers = {
    "Authorization": f"Token {api_key}",
    "Content-Type": "application/json"
}

# Data payload
payload = {
    "url": audio_url,
    "diarize": True
}

# Make the API request
response = requests.post(url, headers=headers, json=payload)

# Print the response
print(response.text)

File: code/community/1306402371709964289/deepgram-diarization-pricing.js
const axios = require('axios');
require('dotenv').config();

// Load environment variables
const apiKey = process.env.DEEPGRAM_API_KEY;
const audioUrl = process.env.AUDIO_URL;

// API endpoint
const url = 'https://api.deepgram.com/v1/listen';

// Request options
const options = {
  headers: {
    'Authorization': `Token ${apiKey}`,
    'Content-Type': 'application/json'
  }
};

const payload = {
  url: audioUrl,
  diarize: true
};

// Make the API request
axios.post(url, payload, options)
  .then(response => {
    console.log(response.data);
  })
  .catch(error => {
    console.error('Error:', error);
  });